drivers/eeprom/eeprom_fm24w256.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2021 Caspar Friedrich <c.s.w.friedrich@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <devicetree.h>
+#include <devicetree/gpio.h>
+#include <drivers/eeprom.h>
+#include <drivers/gpio.h>
+#include <drivers/i2c.h>
+#include <logging/log.h>
+
+#define DT_DRV_COMPAT cypress_fm24w256
+
+// 256-Kbit ferroelectric random access memory logically organized as 32K × 8 bit
+#define SIZE KB(32)
+
+// Get the upper seven address bits and mask them properly
+#define OFFSET_HI(offset) ((offset >> 8) & BIT_MASK(7))
+
+// Get the lower eight address bits and mask them properly. Redundacy for consistency.
+#define OFFSET_LO(offset) ((offset >> 0) & BIT_MASK(8))
+
+LOG_MODULE_REGISTER(fm24w256, CONFIG_EEPROM_LOG_LEVEL);
+
+struct config {
+	struct i2c_dt_spec i2c_spec;
+	struct gpio_dt_spec wp_spec;
+	bool read_only;
+};
+
+static int configure_pin(const struct gpio_dt_spec *spec)
+{
+	if (!spec->port) {
+		return -ENODEV;
+	}
+
+	if (!device_is_ready(spec->port)) {
+		return -ENODEV;
+	}
+
+	return gpio_pin_configure_dt(spec, GPIO_OUTPUT_INACTIVE);
+}
+
+static int init(const struct device *dev)
+{
+	const struct config *config = dev->config;
+	int ret;
+
+	/*
+	 * Write protect via GPIO is optional. Only configure if set.
+	 */
+	if (config->wp_spec.port) {
```

Comment:
You need to check for `device_is_ready()` before using the GPIO device.

Reply:
`fm24w256_configure_pin` checks for that. The idea is that it first checks if the port is not NULL, in that case it is assumed that the pin was not configured and if it is not null checks for readiness:

```static int fm24w256_configure_pin(const struct gpio_dt_spec *spec)
{
	if (!spec->port) {
		return -ENODEV;
	}

	if (!device_is_ready(spec->port)) {
		return -ENODEV;
	}

	return gpio_pin_configure_dt(spec, GPIO_OUTPUT_INACTIVE);
}```